add MicrosoftVostokLogAdapter

ArXa1L opened the pull request:
Hi, our team usually builds projects with `microsoft.aspnetcore.*` and `microsoft.extensions.*` frameworks but API's clients are based on `Vostok.ClusterClient*` library. 

It looks nice but we have to use own adapter
`Microsoft.Extensions.Logging.ILogger --> Vostok.Logging.Abstractions.ILog`.

--
We have an option to create an additional library and provide this adapter but i think it will be useful for others who use `vostok.logging.microsoft` in a similar case.

ArXa1L committed:
add MicrosoftVostokLogAdapter
add new dependency 'vostok.logging.formatting'

kungasc commented on Vostok.Logging.Microsoft/Adapters/MicrosoftVostokLogAdapter.cs:
```diff
@@ -0,0 +1,81 @@
+            $"{{{WellKnownProperties.SourceContext}:w}}" +
+            $"{{{WellKnownTokens.Message}}}");
+
+        public MicrosoftVostokLogAdapter(ILogger logger)
```
1. No need Adapters folder, this whole module about it.
2. Maybe simplify name to VostokLog.
3. Add settings class: `MicrosoftVostokLogAdapter([NonNull] ILogger logger, [CanBeNull] MicrosoftVostokLogAdapter Settings settings)`
https://git.skbkontur.ru/ke/vostok.logging.kontur.legacy/-/blob/master/Vostok.Logging.Kontur.Legacy/KonturLoggingLog.cs#L48

ArXa1L committed:
rename MicrosoftVostokLogAdapter to MicrosoftLog
add MIcrosoftLogSettings
make private field `template` static
rename private method MessageFormatter to FormatMessage

ArXa1L commented:
I rename `MicrosoftVostokLogAdapter` to `MicrosoftLog` instead of `VostokLog` cause it is more appropriate
